aten/src/ATen/native/cpu/Activation.cpp

```diff
@@ -757,12 +762,12 @@ void softshrink_kernel(TensorIteratorBase& iter, const Scalar& lambd) {
     cpu_kernel_vec(
       iter,
       [=](scalar_t a) -> scalar_t {
-        return a > lambd_val ? a - lambd_val : (a < -lambd_val ? a + lambd_val : scalar_t(0));
+        return a > lambd_val ? a - lambd_val : (a < -lambd_val ? a + lambd_val : a * scalar_t(0));
```

Comment:
Why multiply?

Reply:
nan * 0 -> nan. Otherwise 0

Reply:
Can we use nan+0->nan? Because +  is faster.